opencti-platform/opencti-front/src/private/components/common/form/ObjectMarkingField.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useState, FunctionComponent } from 'react';
+import { Field } from 'formik';
+import { graphql } from 'react-relay';
+import makeStyles from '@mui/styles/makeStyles';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogContentText from '@mui/material/DialogContentText';
+import useAuth from '../../../../utils/hooks/useAuth';
+import ItemIcon from '../../../../components/ItemIcon';
+import Transition from '../../../../components/Transition';
+import AutocompleteField from '../../../../components/AutocompleteField';
+import { RenderOption } from '../../../../components/list_lines';
+import { useFormatter } from '../../../../components/i18n';
+import { convertMarking } from '../../../../utils/edition';
+import { Option } from './ReferenceField';
+
+const useStyles = makeStyles(() => ({
+  icon: {
+    paddingTop: 4,
+    display: 'inline-block',
+  },
+  text: {
+    display: 'inline-block',
+    flexGrow: 1,
+    marginLeft: 10,
+  },
+}));
+
+export const objectMarkingFieldAllowedMarkingsQuery = graphql`
+    query ObjectMarkingFieldAllowedMarkingsQuery {
+        me {
+            allowed_marking {
+                id
+                entity_type
+                standard_id
+                definition_type
+                definition
+                x_opencti_color
+                x_opencti_order
+            }
+        }
+    }
+`;
+
+interface ObjectMarkingFieldProps {
+  name: string;
+  style?: React.CSSProperties;
+  onChange: (name: string, values: Option[]) => void;
+  helpertext?: unknown;
+  disabled?: boolean;
+  label?: string;
+  setFieldValue?: (name: string, values: Option[]) => void
+}
+
+const ObjectMarkingField: FunctionComponent<ObjectMarkingFieldProps> = ({
+  name,
+  style,
+  onChange,
+  helpertext,
+  disabled,
+  label,
+  setFieldValue,
+}) => {
+  const classes = useStyles();
+  const { t } = useFormatter();
+  const [newMarking, setNewMarking] = useState<Option[] | undefined>(undefined);
+
+  const { me } = useAuth();
+  const allowedMarkingDefinitions = me.allowed_marking?.map(convertMarking) ?? [];
+
+  const optionSorted = allowedMarkingDefinitions.sort((a, b) => {
+    if (a.entity.definition_type === b.entity.definition_type) {
+      return (a.entity.x_opencti_order < b.entity.x_opencti_order ? -1 : 1);
+    } return (a.entity.definition_type < b.entity.definition_type ? -1 : 1);
```

Comment:
```suggestion
    if (a.definition_type === b.definition_type) {
      return (a.x_opencti_order < b.x_opencti_order ? -1 : 1);
    } return (a.definition_type < b.definition_type ? -1 : 1);
```

Reply:
Yes, entity can be removed now, it was used before some changes :)
(and will be removed from edition.js too)